### PR TITLE
feat: GP2-101 User Import Script

### PR DESCRIPTION
**Pull request by peterstarling**
Example output: 
image

### CODE REVIEW

File: apps/gp2-server/src/data-providers/user.data-provider.ts [hunk 1]
```diff
@@ -105,26 +121,44 @@ export class UserSquidexDataProvider implements UserDataProvider {
...
+
+function mapUserFields(
+  input: UserCreateDataObjectEnumFields,
+): UserCreateInputEnumFields;
+function mapUserFields(
+  input: UserUpdateDataObjectEnumFields,
+): UserUpdateInputEnumFields;
+function mapUserFields(
+  input: UserCreateDataObjectEnumFields | UserUpdateDataObjectEnumFields,
+): UserCreateInputEnumFields | UserUpdateInputEnumFields {
```
Comment (markcsims) [inline apps/gp2-server/src/data-providers/user.data-provider.ts:147]: I know it's doing the overloading but I found that very difficult to read. 

Also, we've lost the destructuring, is this intentional?

File: apps/gp2-server/src/handlers/webhooks/fetch-by-code-handler.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
 import { ValidationError } from '@asap-hub/errors';
 import { validateAuth0Request } from '@asap-hub/server-common';
 import { framework as lambda } from '@asap-hub/services-common';
-import { RestUser, SquidexGraphql, SquidexRest } from '@asap-hub/squidex';
+import {
+  gp2 as gp2squidex,
```
Comment (markcsims) [inline apps/gp2-server/src/handlers/webhooks/fetch-by-code-handler.ts:5]:
```suggestion
  gp2 as gp2Squidex,
```

File: apps/gp2-server/test/data-providers/user.data-provider.test.ts [hunk 3]
```diff
@@ -290,6 +301,114 @@ describe('User data provider', () => {
...
+
+    test.each`
+      region                      | expected
+      ${'Africa'}                 | ${UsersDataRegionEnum.Africa}
+      ${'Asia'}                   | ${UsersDataRegionEnum.Asia}
+      ${'Australia/Australiasia'} | ${UsersDataRegionEnum.AustraliaAustraliasia}
+      ${'Europe'}                 | ${UsersDataRegionEnum.Europe}
+      ${'North America'}          | ${UsersDataRegionEnum.NorthAmerica}
+      ${'South America'}          | ${UsersDataRegionEnum.SouthAmerica}
+      ${'Latin America'}          | ${UsersDataRegionEnum.LatinAmerica}
```
Comment (markcsims) [inline apps/gp2-server/test/data-providers/user.data-provider.test.ts:344]: We shouldn't use the GraphQL Enum. We've typed them again in the squidex package.

File: apps/gp2-server/test/data-providers/user.data-provider.test.ts [hunk 4]
```diff
@@ -290,6 +301,114 @@ describe('User data provider', () => {
...
+      },
+    );
+
+    test.each`
+      role                           | expected
+      ${'Working Group Participant'} | ${UsersDataRoleEnum.WorkingGroupParticipant}
+      ${'Network Investigator'}      | ${UsersDataRoleEnum.NetworkInvestigator}
+      ${'Network Collaborator'}      | ${UsersDataRoleEnum.NetworkCollaborator}
+      ${'Administrator'}             | ${UsersDataRoleEnum.Administrator}
+      ${'Trainee'}                   | ${UsersDataRoleEnum.Trainee}
```
Comment (markcsims) [inline apps/gp2-server/test/data-providers/user.data-provider.test.ts:370]: See above.

File: apps/gp2-server/test/data-providers/user.data-provider.test.ts [hunk 5]
```diff
@@ -290,6 +301,114 @@ describe('User data provider', () => {
...
+          .reply(200, fetchUserResponse());
+
+        await userDataProvider.create({
+          ...userCreateDataObject,
+          role,
+        });
+      },
+    );
+
+    test.each(Object.values(UsersDataDegreeEnum))(
```
Comment (markcsims) [inline apps/gp2-server/test/data-providers/user.data-provider.test.ts:390]: See above

File: apps/gp2-server/test/fixtures/user.fixtures.ts [hunk 6]
```diff
@@ -1,6 +1,10 @@
 import { gp2 } from '@asap-hub/model';
 import { UserEvent } from '@asap-hub/server-common';
-import { gp2 as SquidexGp2, WebhookPayload } from '@asap-hub/squidex';
+import {
+  gp2 as SquidexGp2,
```
[the remark this answers is not in the document]
Comment (markcsims) [reply apps/gp2-server/test/fixtures/user.fixtures.ts:4]: maybe we should rename the model from gp2 to gp2Model. We've done this in other instances.

File: apps/gp2-server/scripts/import-users.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+  const args = process.argv.slice(2);
+
+  if (typeof args[0] !== 'string') {
+    throw new Error('Please provide a path to the CSV file');
+  }
+
+  const filePath = args[0];
+
+  const a = parse(
```
Comment (markcsims) [inline apps/gp2-server/scripts/import-users.ts:48]: we should look for a more descriptive name.

Comment (peterstarling) [inline apps/gp2-server/scripts/import-users.ts:48]:
```suggestion
  const userCsvImport = parse(
```

File: apps/gp2-server/scripts/import-users.ts [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+          }
+        }
+
+        usersFailed++;
+        console.log({ input, e });
+      }
+    },
+  );
+
+  await a(filePath);
```
Comment (peterstarling) [inline apps/gp2-server/scripts/import-users.ts:94]:
```suggestion
  await userCsvImport(filePath);
```

**Review by markcsims — APPROVED** [review]

**Pull request merged**